aiobotocore/endpoint.py

```diff
@@ -39,7 +39,9 @@ def text_(s, encoding='utf-8', errors='strict'):
 @asyncio.coroutine
 def convert_to_response_dict(http_response, operation_model):
     response_dict = {
-        'headers': http_response.headers,
+        # botocore converts keys to str, so make sure that they are in
+        # the expected case.
+        'headers': {k.lower(): v for k, v in http_response.headers.items()},
```

Comment:
`Location` does not go lowercase, S3 simply does not provide this header. 

And yes, the issue is that dict()'ing a request response will not cause the headers to go lowercase, while `botocore` expects them to be lowercased